notifications_utils/clients/redis/redis_client.py

```diff
@@ -90,24 +94,31 @@ def increment_hash_value(self, key, value, raise_exception=False, incr_by=1):
             try:
                 return self.redis_store.hincrby(key, value, incr_by)
             except Exception as e:
-                self.__handle_exception(e, raise_exception)
+                self.__handle_exception(e, raise_exception, 'increment_hash_value', key)
 
     def get_all_from_hash(self, key, raise_exception=False):
         if self.active:
             try:
                 return self.redis_store.hgetall(key)
             except Exception as e:
-                self.__handle_exception(e, raise_exception)
+                self.__handle_exception(e, raise_exception, 'get_all_from_hash', key)
 
     def set_hash_and_expire(self, key, values, expire_in_seconds, raise_exception=False):
         if self.active:
             try:
                 self.redis_store.hmset(key, values)
                 return self.redis_store.expire(key, expire_in_seconds)
             except Exception as e:
-                self.__handle_exception(e, raise_exception, "Error setting key: {} in redis cache".format(key))
+                self.__handle_exception(e, raise_exception, 'set_hash_and_expire', key)
+
+    def expire(self, key, expire_in_seconds, raise_exception=False):
+        if self.active:
+            try:
+                self.redis_store.expire(key, expire_in_seconds)
+            except Exception as e:
+                self.__handle_exception(e, raise_exception, 'expire', key)
 
-    def __handle_exception(self, e, raise_exception, message=None):
-        current_app.logger.exception("{} \n {}".format(message, e))
+    def __handle_exception(self, e, raise_exception, operation, key_name):
+        current_app.logger.exception('Redis error performing {} on {}'.format(operation, key_name))
         if raise_exception:
             raise e
```

Comment:
Are all exceptions likely to be RedisError?
I'd usually go for raising a custom exception (Error) here i.e. RedisClientError.

As a consumer I would want to easily catch exceptions from a library that would encapsulate RedisError but also any other error e.g. TypeError etc

Reply:
yeah our wrapper is super simple and I've never seen anything other than a `RedisError` thrown - if there's something else i'd probs want us to investigate it as a bug anyway.